hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancerSelectionCriteria.java

```diff
@@ -158,12 +155,12 @@ private Comparator<ContainerID> orderContainersByUsedBytes() {
 
   /**
    * Checks whether a Container has the ReplicationType
-   * {@link HddsProtos.ReplicationType#EC}.
+   * {@link HddsProtos.ReplicationType#EC} and the Legacy Replication Manger is enabled.
    * @param container container to check
    * @return true if the ReplicationType is EC and "hdds.scm.replication
    * .enable.legacy" is true, else false
    */
-  private boolean isECContainer(ContainerInfo container) {
+  private boolean isECContainerAndLegacyRMEnabled(ContainerInfo container) {
```

Comment:
Maybe it's just me, but this suggests to me as if `Enabled` applies to both legacy RM and EC container.

How about `isUnsupportedType` (or similar), to hide internal details?